backend/src/packages/shifts/shift.repository.ts

```diff
@@ -40,6 +43,55 @@ class ShiftRepository implements IRepository {
       .execute();
   }
 
+  public async getOpenedByDriver(
+    driverId: number,
+  ): Promise<ShiftDatabaseModel | null> {
+    const [shift = null] = await this.db
+      .driver()
+      .select()
+      .from(this.shiftSchema)
+      .where(
+        and(
+          isNull(this.shiftSchema.endDate),
+          eq(this.shiftSchema.driverId, driverId),
+        ),
+      )
+      .execute();
+
+    return shift;
+  }
+
+  public async getOpenedByTruckWithBusiness(
+    truckId: number,
+  ): Promise<
+    | (ShiftDatabaseModel & { businessId: number; driverLicenseNumber: string })
+    | null
+  > {
+    const [shift = null] = await this.db
+      .driver()
+      .select()
+      .from(this.shiftSchema)
+      .innerJoin(
+        schema.drivers,
+        eq(this.shiftSchema.driverId, schema.drivers.userId),
```

Comment:
```suggestion
        eq(this.shiftSchema.driverId, this.driversSchema.userId),
```
and change it in all other places